py_ecc/fields/optimized_field_elements.py

```diff
@@ -0,0 +1,395 @@
+from typing import (  # noqa: F401
+    cast,
+    List,
+    Sequence,
+    Tuple,
+    Union,
+)
+
+
+from py_ecc.fields.field_properties import (
+    field_properties,
+)
+
+from py_ecc.utils import (
+    deg,
+    prime_field_inv,
+)
+
+
+IntOrFQ = Union[int, "FQ"]
+
+
+class FQ(object):
```

Comment:
Looks like we can just delete this `FQ` and import it from the regular (unoptimized) implementation.

Reply:
@carver I have observed this prior, but have just copy/pasted because to maintain symmetry between the `optimized` and the `non-optimized` version in terms of `type hinting`, `imports` etc at this point of time.

How about I change this after we complete this whole refactor thing?